### PR TITLE
Provide with options in build.cmd

### PR DESCRIPTION
**Pull request by qinezh**
```
build template    => only update template, no build project
build raw         => build except updating template
build debug       => build target debug
build release     => build target release
build prod        => update version before build
build             => update template before build
```

### CODE REVIEW

**Comment by qinezh** [comment]
@chenkennt @vwxyzh @superyyrrzz @hellosnow @ansyral

**Comment by vwxyzh** [comment]
👍

**Comment by superyyrrzz** [comment]
:+1:

File: build.cmd [hunk 1]
```diff
@@ -4,51 +4,96 @@ PUSHD %~dp0
...
-:: Check if DNU exists globally
-:: DNU is OPTIONAL
-WHERE dnu >NUL
-IF NOT '%ERRORLEVEL%'=='0' (
-    ECHO ERROR: DNU is not successfully configured.
-    ECHO ERROR: Please follow http://docs.asp.net/en/latest/getting-started/installing-on-windows.html#install-the-net-version-manager-dnvm to install dnvm.
-    ECHO ERROR: If dnvm is installed, use `dnvm list` to show available dnx runtime, and use `dnvm use` to select the default dnx runtime
-    GOTO :Exit
+IF /I [%UpdateTemplate%]==[template] (
+    CALL UpdateTemplate.cmd
```
Comment (hellosnow) [inline build.cmd]: goto UpdateTemplate?

File: build.cmd [hunk 2]
```diff
@@ -4,51 +4,96 @@ PUSHD %~dp0
...
+    GOTO Next
+)
+
+if /I [%1]==[PROD] (
+    SET Environment=PROD
+    GOTO Next
+)
+
+if /I [%1]==[raw] (
+    SET UpdateTemplate=%1
```
Comment (hellosnow) [inline build.cmd]: question: why use UpdateTemplate for build raw, which is not update templates?

File: build.cmd (hunk 1 above)
Comment (qinezh) [inline build.cmd]: in this way, it not only updates template but also build project.

---
In reply to: [68713565](https://github.com/dotnet/docfx/pull/435#discussion_r68713565) [](ancestors = 68713565)

File: build.cmd (hunk 2 above)
Comment (qinezh) [inline build.cmd]: build template => only update template, not build project

build raw => not update template, only build project

---
In reply to: [68713739](https://github.com/dotnet/docfx/pull/435#discussion_r68713739) [](ancestors = 68713739)

Comment (hellosnow) [inline build.cmd]: that's what I mean, so "build raw" shouldn't named with "UpdateTemplate"? how about "SkipTemplate"? Also update line 73 error msg

---
In reply to: [68714739](https://github.com/dotnet/docfx/pull/435#discussion_r68714739) [](ancestors = 68714739,68713739)

File: build.cmd (hunk 1 above)
Comment (hellosnow) [inline build.cmd]: got it. Also "ECHO Updating template" before this line?

---
In reply to: [68714523](https://github.com/dotnet/docfx/pull/435#discussion_r68714523) [](ancestors = 68714523,68713565)

File: build.cmd (hunk 2 above)
Comment (qinezh) [inline build.cmd]: greate advice!

---
In reply to: [68715890](https://github.com/dotnet/docfx/pull/435#discussion_r68715890) [](ancestors = 68715890,68714739,68713739)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**